### DISCUSSION TITLE
Can't deploy to Vercel - Restarted collecting page data for [object Object] because it took more than 60 seconds

### DISCUSSION THREAD

**Discussion by chen-rn** [vercel/next.js#52280] [App Router]
The build works locally but can't get it to work on Vercel.
Here's the log
```
Static worker unexpectedly exited with code: null and signal: SIGSEGV
--
01:08:46.511 | - warn Restarted collecting page data for [object Object] because it took more than 60 seconds
01:08:46.512 | - warn See more info here https://nextjs.org/docs/messages/static-page-generation-timeout
01:08:47.184 | Static worker unexpectedly exited with code: null and signal: SIGSEGV
01:09:46.896 |  
01:09:46.896 | > Build error occurred
01:09:46.898 | Error: Collecting page data for [object Object] is still timing out after 2 attempts. See more info here https://nextjs.org/docs/messages/page-data-collection-timeout
01:09:46.898 | at onRestart (/vercel/path0/node_modules/next/dist/build/index.js:701:39)
01:09:46.898 | at Worker.isPageStatic (/vercel/path0/node_modules/next/dist/lib/worker.js:88:40)
01:09:46.898 | at processTicksAndRejections (node:internal/process/task_queues:96:5)
01:09:46.898 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:103:20)
01:09:46.898 | at async /vercel/path0/node_modules/next/dist/build/index.js:861:56
01:09:46.898 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:103:20)
01:09:46.898 | at async Promise.all (index 5)
01:09:46.898 | at async /vercel/path0/node_modules/next/dist/build/index.js:796:17
01:09:46.898 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:103:20)
01:09:46.899 | at async /vercel/path0/node_modules/next/dist/build/index.js:735:129
01:09:46.950 | Error: Command "npm run build" exited with 1
```

I'm not using any getServerSideProps(directly at least)

**I've tried:**
- upgrading all packages
- node 16/18
- changing timeout duration to 1000

**Comment by icyJoseph** [comment]
Are you using any other data fetching function? I've seen this before, happening locally, and yeah it is annoying that it is printing `[object Object]` - 

I wonder if you could remove pages and push until you get a build running, sort of searching for the culprit page.

**Reply by icyJoseph** [reply]
Are you using app router?

**Reply by icyJoseph** [reply]
I tried a few things and nothing trigger this issue :/ 

I wonder if you could give me some hints on how to reproduce it. Maybe it is a 404 page, or error page that's triggering the problem?

**Reply by chen-rn** [reply]
Hey! yes I'm using app router.

I ended up creating another github repo, copy pasted my code over, and the deployment works. In retrospect, I should've just removed pages one by one like you mentioned, but it wasn't clear to me that it was a data fetching issue.

However, one of the pages would cause a timeout when I navigate there, and I imagine it's the same underlying issue.
The issue was using [youtube.js](https://github.com/LuanRT/YouTube.js) to fetch data.

For some reason, the deployment works on Railway, and I have still yet to figure out why.

**Comment by Enalmada** [comment]
I just experienced this and it ended up being an app router api route explicitly marked dynamic but build was trying to do the "Collecting page data" on it anyway.  A file deep in the specific route logic had some code in it doing some that worked fine in dev but the build didn't like.  Fairly frustrating how much file deleting I had to do to figure out what the [object Object] was.  

```
if (env.APP_ENV === 'local') {
    const fs = await import('fs/promises');
    const path = await import('path');
    const graphql = await import('graphql');
    const schemaAsString = graphql.printSchema(graphql.lexicographicSortSchema(schema));
    ...
}
```

**Comment by maxijonson** [comment]
I'm getting a similar, but instead of `[Object object]`, I get `undefined`:

```
Collecting page data for undefined is still timing out after 2 attempts.
```

This error is super hard to debug... we have so many pages using `generateStaticParams` and there's no way of knowing which page is the source of the timeout. The last time we had this error, it ended up being because `generateStaticParams` was returning an array of numbers `[1, 2, 3]` instead of `[{ page: 1 }, { page: 2}, { page: 3 }]`, which is obviously wrong because you need to return props as an object. However, even then, we had to go through all of our pages and find the culprit. There were no hints or obvious error message saying that this one was returning the wrong data type.
